### ISSUE TITLE
[bug]: Fresh Install stuck in `Waiting for system cert key file...`

### ISSUE THREAD

**Issue by storopoli** [Start9Labs/cln-startos#130]
### Prerequisites

- [X] I have searched for existing issues that already report this problem.

### Device

Laptop/Desktop

### Device OS

Other

### Device OS Version

startos

### Browser

Firefox

### Browser Version

127

### Current Behavior

Installed CLN from scratch and now it is stuck in

`Waiting for system cert key file...`

I did a search and found that I am hitting this line: https://github.com/Start9Labs/cln-startos/blob/29f0fe7924543e7689026ad0194cd3beb38a1048/docker_entrypoint.sh#L106-L111

I have no idea what's going on. My start9 `/mnt` dir does not have anything.
Shouldn't CLN generate a system cert key file?

Cc @chrisguida since 48e6bbbf5d3c743f5c701567f6aa2b1dd9981e3d introduce these lines.

### Expected Behavior

CLN should start from a fresh install in StartOS and not be stuck in a infinite loop.

### Steps to Reproduce

1. Install CLN from scratch
2. Run and see the logs

### Anything else?

![image](https://github.com/Start9Labs/cln-startos/assets/43353831/12747b29-bdcc-4e31-83dc-9ad98dd9dfcd)

**Comment by storopoli** [comment]
Update, after fiddling around I've discovered that the cert directory in startos is:

`pub const PACKAGE_CERT_PATH: &str = "/var/lib/embassy/ssl";` (https://github.com/Start9Labs/start-os/blob/fc8b1193de618efe3c9fe9f68ed9c7ce23cd562f/core/startos/src/net/mod.rs#L23).

Then:

```bash
cp /embassy-data/package-data/volumes/c-lightning/data/main/bitcoin/server-key.pem /var/lib/embassy/ssl/c-lightning/rest/rest.key.pem
cp /embassy-data/package-data/volumes/c-lightning/data/main/bitcoin/server.pem /var/lib/embassy/ssl/c-lightning/rest/rest.cert.pem
```

seems to fix.
Maybe the `docker_entrypoint.sh` should be made robust to that kind of scenario?

**Comment by chrisguida** [comment]
@storopoli I've stopped maintaining this package, but I'm sure @Dominion5254 would accept a PR to fix :)

**Comment by chrisguida** [comment]
FTR this is probably an edge case on your particular hardware, I haven't ever seen this on any of my installs. But yeah I'm sure my code is not watertight :)

**Comment by storopoli** [comment]
Start9 Server One 2023 with Celeron N4505
Version 0.3.5~1

This started happening after I've `chroot`ed and created some `systemd` `socat` services following https://community.start9.com/t/core-lightning-with-tor-and-ipv4-clearnet/965

I've also added the following lines to `config.main`:

```
announce-addr=<CLEARNET_DNS_DOMAIN>:9735
announce-addr-dns=true
```

**Comment by chrisguida** [comment]
Ahh yeah that's definitely a hack. I can't immediately think of a reason that would break the container, but doing stuff like that definitely 'voids your warranty" :p

Have you tried just turning that off and seeing if it works? The container is not expecting the `config.main` file to be changed by the user.

**Comment by storopoli** [comment]
Yes tried restoring from a backup, wiping the `/embassy-data/.../c-lightning` folder. Tried from a brand new node.
Unfortunately I was forced to move to LND after 1 solid year in CLN.
I will probably wait for 0.3.6 that will arrive in $[14, \infty)$ days.

Feel free to close this if you cannot reproduce it.

**Comment by chrisguida** [comment]
It's doing the same thing from a fresh node with nothing in the datadir? That doesn't make any sense...

**Comment by storopoli** [comment]
Yes, because it does not have the `/mnt/cert/` folder somehow

**Comment by Dominion5254** [comment]
The direct cause isn't immediately apparent to me either, but the clearnet hack almost certainly seems to be the culprit for the original issue. While it is great to see users doing cool DIY hacks, going under the hood like to make changes such as this or resolving resulting issues is of course not something Start9 can officially support.

But I am confused by why this would occur with "a brand new node" - by this do you mean re-installing CLN on the same box that you had `chroot`ed, or a completely fresh install of StartOS?

**Comment by chrisguida** [comment]
I mean this is an OS issue if the cert volume isn't getting mounted on a fresh new install of CLN on a brand new start-os server.

**Comment by Dominion5254** [comment]
Agreed, if that is the case it warrants opening an issue on the StartOS repo, but I want to make sure that is in fact what @storopoli meant by "a brand new node".

**Comment by MattDHill** [comment]
No longer applicable — `docker_entrypoint.sh` was removed when the package was rewritten for the 0.4.0 SDK, and CLN no longer waits on a system cert key file.
